CHANGELOG.md

```diff
@@ -49,6 +49,20 @@ you should edit it (`kubectl edit validatingwebhookconfiguration
 kong-validations`) and add `kongclusterplugins` under the `resources` block for
 the `configuration.konghq.com` API group.
 
+#### Breaking changes
+
+- You must upgrade to 2.0.x before upgrading to 2.1.x to properly handle the
+  transition from apiextensions.k8s.io/v1beta1 CRDs to apiextensions.k8s.io/v1
+  CRDSs. CRDs are now generated from their underlying Go structures to avoid
+  accidental mismatches between implementation and Kubernetes configuration.
+  KongIngresses previously included `healthchecks.passive.unhealthy.timeout`
+  and `healthchecks.active.unhealthy.timeout` fields that did not match the
+  corresponding Kong configuration and had no effect. These are now
+  `healthchecks.passive.unhealthy.timeouts` and
+  `healthchecks.active.unhealthy.timeouts`, respectively. If you use these
+  fields, you must rename them in your KongIngresses before upgrading.
+  [#1971](https://github.com/Kong/kubernetes-ingress-controller/pull/1971)
```

Comment:
It seems the entire reason we have this language is because we still have some uncertainty around upgrading the old `v1beta1` CRDs to these latest `v1` CRDs. After thinking this over, it doesn't feel right to me to put this burden on the end-users. The fact that the generator hack which added an explicit `false` is not needed any more suggests that something has changed in upstream since this was last tested, or we missed something.

Reply:
I suppose I'm just very on the fence about putting this burden on the end-user when we've had an existing a workaround that would avoid the need to have this upgrade jump. I would like to hear from @mflendrich on this to see what his thoughts are :thinking:

Reply:
Two thoughts from my side.

1. With such a cross-cutting change as a CRD update, it feels reasonable to expect that the user upgrade via a bridge version. This also limits the potential for corner case bugs going forward (as any future chart upgrades will need to be compatible only with upgrades from 2.0 onwards)
2. if my understanding is correct, this rule (that going through 2.0.x is a must) also frees us of the need to maintain the hack for `preserveUnknownFields` anymore. That's a good thing maintenance-wise.

I also feel the motivation to make the experience to the user as smooth as possible. Here's my assessment of the situation, though:
- we (the maintainers) have limited resources
- deleting the preserveUnknownFields hack is a win for code health
- hopping through 2.0.x greatly simplfiies the constellation of versions we'll need to support upgrades from, which is a win for maintenance cost, at the expense of one more operation a user needs to perform

I wish that this change (to lift `preserveUnknownFields`) was isolated in a separate PR - that would greatly simplify review - but otherwise I think it's a change for good.

Reply:
:+1: